README.md

# R Data package for Lake Forest Park StreamKeepers

See setupandrun.R for rough usage examples.

See http://www.lfpstreamkeepers.org for more details.

# Things to Do
- document Quality data update  procedure
- document BIBI data update procedure
- document data sets, updating ddatasets
- think about using githubs issue tracker for this things to do list
- update package readme to be package use and maintainence document
- figure out where the 2013 BIBI data went.
- convert loadBIBI.R into a function.


LFPStreamKeepersData/NEWS
NEWS/Change Log for LFPStreamKeepersData package
------------------------------------------------

0.3-2 2015-01-17
      o Added winter 2015 Quality Data
      o Added Fall 2014 BIBI Data spreadsheets
      o Added Fall 2014 BIBI Data to bibiSummaryData.csv

0.3-1 2014-10-21
      o Added summer 2014 Quality Data

0.3-1 2014-01-18
      o Added winter 2014 Quality Data

0.3-0 2014-01-16
      o Added 2013 BIBI Summary data
      o DESCRIPTION file gets URL

0.2-1 2013-09-29
        o Added Site.barplot to NAMESPACE, making it an
	  externally visible function.
	o Updated title in plots produced by Site.barplot.
	o Gave Site.barplots a legend explaining total score.
	o SetUpAndRun.R has section to produce plots for website.

0.2-0 2013-07-20
        o New Data from July Testing.
	o Add 2006 - 2011 to rawQualityData.csv
	o Moved SetUpAndRun.R out of package because it
          operates directly on package.
	o Added XLConnect, rJava to Recommended because
	  loadBIBI requires them.
        o export lfpskLoadQualityData.

0.1-6 2013-03-16
        o Tweaked Site.barplot() to put legend on first bar.

0.1-5 2013-03-16
        o Tweaked Site.barplot() to allow legend to work.

0.1-4 2013-02-08
	o plotBIBI.r - added Site.barplot() function

0.1-3 2013-01-29
    o Added 1/19/2013 data fto rawQualityData.csv
	  - Dropped ratings columns because they shouuld be calculated.
	  - Dropped Temp Celcius column because it can be calculated.
	  - Added Turbidity column: Need to go through old data and fix
	    sky and turbidity data.

0.1-2 2013-01-25
	o Made BIBI data visible to the Revo Project.
	o Normalizing rawQualityData.csv
	  Removed aprostophes (single quotes) from reach names,
	  Removed trailing commas.

0.1-1 2013-01-23
    o Added rawQualityData.csv, skeliton of data set documentation.
      rawQualityData.csv needs cleaning: read.table doesn't like it.

0.1-0 2013-01-16
    o loadBIBI.r loads all of the BIBI scoring data into bibiScoringData
      object.  Now ready to start working on plotting this data.
    o Added 2008 BIBI data.

0.0-5 2013-01-14
    o Added Replicate column to readScoringXLS, assigned replicate values.
    o Added readScoringXLS to namespace.

0.0-4 2013-01-13
    o generalized LyonFiene2007 load data function to
      Added readScoringXLS load data function.

0.0-3 2013-01-11
    o Added Lyon Fiene 2007 load data function

0.0-2 2012-12-23
    o Adding Benthic Details Data.

0.0-1 2012-12-23
    o Initial Version
    o LFP StreamKeepers bibiSummaryData.csv


SetupAndRun.R
gittop <- "c:/Users/Derek/Documents/github/lfpstreamkeepers"
setwd( gittop )
dir()
getwd()
system("R CMD build LFPStreamKeepersData")
install.packages("XLConnect", lib="c:/users/derek/R_LIBS_USER")
install.packages("XLConnect")
# note that XLConnect requires Java
detach("package:LFPStreamKeepersData",unload=TRUE)
install.packages("LFPStreamKeepersData_0.3-2.tar.gz", type="source", lib="C:/users/derek/R_LIBS_USER" )
search()

###############################
## Update bibi data, given new spread sheets
## 1. Check spread sheets in under new year folder in
##    inst/data/BIBI/
## add new year to loadBIBI.R
## Rebuild package

library( LFPStreamKeepersData )

y2014= list(
lyon35th =       readScoringXls( "extdata/BIBI/2014/Lyon\ at\ 35th\ 2014\ Scoring.xlsx" ),
lyon178th =      readScoringXls( "extdata/BIBI/2014/Lyon\ crk\ at\ 178th\ 2014\ Scoring.xlsx" ),
mcAleerAcres =   readScoringXls( "extdata/BIBI/2014/McAleer\ Crk\ at\ Animal\ Acres\ 2014\ Scoring.xlsx" ),

mcAleerPerkins = readScoringXls( "extdata/BIBI/2014/McAleer\ Crk\ at\ Perkins\ 2014\ Scoring.xlsx" )
)


###############################
# Produce png's for website.
library( LFPStreamKeepersData )

plotBIBI( system.file("extdata", "BIBI", 'bibiSummaryData.csv', package="LFPStreamKeepersData"),createJpg=FALSE )

source(system.file("loadBIBI.R", package="LFPStreamKeepersData"))
source("LFPStreamKeepersData/R/plotBIBI.R")

# plot em to graphics device
Site.barplot(CollectionSite = "mcAleerAcres")
Site.barplot(CollectionSite = "mcAleerPerkins")
Site.barplot(CollectionSite = "lyon178th")
Site.barplot(CollectionSite = "lyon35th")

# grab the graphics device and put it in a png.
plotOne <- function (CollectionSite) {
    cat( CollectionSite,"\n" )
    Site.barplot(CollectionSite = CollectionSite)
    par(bg="white")
    dev.copy(png,filename=paste( CollectionSite,".bibiScore.png", sep=""),
             height=600, width=800)
    dev.off()
}
z <- lapply( c("mcAleerAcres", "mcAleerPerkins", "lyon178th", "lyon35th"), FUN=plotOne )

Update webpage with these plots Noting they are
essentially a graphical representation of the composite table.

######
# data accessor: load the raw data.
library( LFPStreamKeepersData )
source("LFPStreamKeepersData/R/plotBIBI.R")
source("LFPStreamKeepersData/inst/loadBIBI.R")

bibiScoringData
names(bibiScoringData)
names(bibiScoringData$y2012)
nil

################
## Connect to Excel File Pull and Format Data
# This one seems to drop data.
library(RODBC)
excel.connect <- odbcConnectExcel(system.file("extdata/BIBI/2006/Lyon_at_fiene_scoring_2006.xls",package="LFPStreamKeepersData"))
sheet.name <- "Sheet1"
dat <- sqlFetch(excel.connect, sheet.name, na.strings=c("","-"))
odbcClose(excel.connect)

#################

lyonFiene2006 <- readScoringXls( "extdata/BIBI/2006/Lyon_at_fiene_scoring_2006.xls")
lyonFiene2007 <- readScoringXls( "extdata/BIBI/2007/Lyon_Creek_Fiene_Scoring_2007.xls" )
##################
options(width=180)

##################
source("R/plotBIBI.R")
Site.barplot()
##################
library( LFPStreamKeepersData )
options(width=100)
qd <- lfpskLoadQualityData()
head(qd)
winter2013 <- subset(qd, Date == "1/19/2013" )
write.table(winter2013)

print(xtable(winter2013), type="html")


x <- as.matrix(format(winter2013))
rownames(x) <- rep("", nrow(x))
print(x, quote=FALSE, right=TRUE)

########
# format quality data for web.
# Used to produce tables found at:
# http://www.lfpstreamkeepers.org/data/quality-data-2013
library( LFPStreamKeepersData )
options(width=100)
qd <- lfpskLoadQualityData()
qualData <- as.matrix(format(subset(qd, Date == " 1/17/2015" )))
rownames(qualData) <- rep("", nrow(qualData))
print(qualData, quote=FALSE, right=TRUE)

qd

########
# format BIBI data for stacked area plot generated by Tableau Software
# Used to produce charts found at:
# ???
library( LFPStreamKeepersData )
options(width=100)
source(system.file("loadBIBI.R", package="LFPStreamKeepersData"))
names(bibiScoringData$y2006)

siteYear <- function( site, year ) {
    df1 <- eval(parse(text=paste(sep="","bibiScoringData$y", year, "$", site, "[bibiScoringData$y", year, "$", site, "$Replicate=='Composite',c(1,3)]" ) ))
    df1["Year"] <- year
    df1
}

for (site in c( "lyon35th", "lyon178th", "mcAleerAcres", "mcAleerPerkins")) {

    df = data.frame("Metric"=character(),"Year"=numeric(),"Score"=numeric())
    df = rbind( df, siteYear( site, 2006 ))
    df = rbind( df, siteYear( site, 2007 ))
    df = rbind( df, siteYear( site, 2008 ))
    df = rbind( df, siteYear( site, 2009 ))
    df = rbind( df, siteYear( site, 2010 ))
    df = rbind( df, siteYear( site, 2012 ))

    df[,c("Score")] = sapply( df[,c("Score")],as.numeric)
    df = df[c("Metric","Year","Score")]

    write.table(df,file=paste(sep="",site,".txt"),sep="\t",row.names=FALSE)
#write.table(df,file="mcAleerAcres.txt",sep= "\t",row.names=FALSE)
}
